### ISSUE TITLE
Incorporate mg-line-brushing extension directly into metricsgraphics

### ISSUE THREAD

**Issue by wlach** [mozilla/metrics-graphics#790]
@dandehavilland wrote a great extension to metrics graphics that lets you brush through line charts and get an expanded view of them. You can see an example of it in action on the current metricsgraphics site:

https://www.metricsgraphicsjs.org/examples.htm#addons

Unfortunately for those of us installing metricsgraphics via npm, this package is not there. I think it likely makes sense to incorporate this directly into metricsgraphics itself, as I would consider this core functionality for a useful charting library.

**Comment by dandehavilland** [comment]
That's already available via npm: https://www.npmjs.com/package/mg-line-brushing

I just verified that `npm i mg-line-brushing` works for me.

**Comment by wlach** [comment]
Oh that's great. Longer term, what do you think about including this plugin in metricsgraphics itself?

**Comment by wlach** [comment]
Part of my rationale is that I eventually want to support more sophisticated types of brushing, along with zooming -- and I think this will be easier to implement inside metricsgraphics itself.

**Comment by dandehavilland** [comment]
Yeah, I'd be glad for it to become a part of MG proper. It's been dormant for a while so it'll no doubt need some tweaking. 👍

**Comment by hamilton** [comment]
All for this.

**Comment by shikhar-scs** [comment]
@wlach Can I work on this issue under your guidance please ?

**Comment by wlach** [comment]
@shikhar-scs absolutely! that would be great if you wanted to take this up.

**Comment by wlach** [comment]
I think we're just going to move the code that's currently in the plugin inline into metricsgraphics. Probably the best way to start is to look at the plugin source https://github.com/dandehavilland/mg-line-brushing/blob/master/src/js/hooks.js and try to create a mental model of how it works. From there, it should be possible to come with a strategy on how to integrate it. If you get stuck, let me know and I can provide some hints.

**Comment by shikhar-scs** [comment]
Ok @wlach I'll get started with this .

> If you get stuck, let me know and I can provide some hints.

Yeah sure 👍

**Comment by shikhar-scs** [comment]
@wlach I guess I have solved the problem. I'll soon file a PR for the same

**Comment by cnwangjie** [comment]
I submitted a pull request #851 with my attempt and some of my questions. Can you take a look and give me some advice? @wlach
